### PR TITLE
Highlighted Names [PORTED FROM POLARIS]

### PR DESCRIPTION
**Pull request by 77seven77seven**
# ORIGINAL PR: https://github.com/PolarisSS13/Polaris/pull/3080

## About the Pull Request

A modified version of the original Polaris PR, if you're mentioned in radio the message is bigger.
This is an optional feature! It can be turned off in the Preferences tab.

## Why It's Good For The Game

Nice QoL feature, and easier for AIs and COs alike to know when someone needs their attention.

image

<sup><sub>AI door!</sub></sup>

## Authorship

https://github.com/Neerti

## Changelog

:cl:
rscadd: If your name is mentioned in IC radio chat it gets highlighted, neat for AIs and COs! (This can be toggled off in the preferences tab.)
/:cl:

### CODE REVIEW

File: code/modules/mob/hear_say.dm [hunk 1]
```diff
@@ -68,8 +70,9 @@
 			else if(!is_blind())
 				to_chat(src, "<span class='name'>[speaker_name]</span>[alt_name] talks but you cannot hear \him.")
 	else
+		var/message_to_send = null
+		var/nverb = verb
```
Comment (EgorDinamit) [inline code/modules/mob/hear_say.dm:74]: There's no reasons to move it

File: code/modules/mob/hear_say.dm [hunk 2]
```diff
@@ -83,22 +86,71 @@
 							nverb = "[verb] ([language.shorthand])"
 						if(GLOB.PREF_OFF)//Regular output
 							nverb = verb
-			on_hear_say("<span class='game say'>[track]<span class='name'>[speaker_name]</span>[alt_name] [language.format_message(message, nverb)]</span>")
+			message_to_send = "<span class='game say'><span class='name'>[speaker_name]</span>[alt_name] [track][language.format_message(message, nverb)]</span>"
```
Comment (EgorDinamit) [inline code/modules/mob/hear_say.dm]: Why did you move the tracking link?

File: code/modules/mob/hear_say.dm [hunk 3]
```diff
@@ -83,22 +86,71 @@
...
+			message_to_send = "<span class='game say'><span class='name'>[speaker_name]</span>[alt_name] [track][nverb], <span class='message'><span class='body'>\"[message]\"</span></span></span>"
+			if(check_mentioned(message) && get_preference_value(/datum/client_preference/check_mention) == GLOB.PREF_YES)
+				message_to_send = "<font size='3'><b>[message_to_send]</b></font>"
+
+		on_hear_say(message_to_send)
 		if (speech_sound && (get_dist(speaker, src) <= world.view && src.z == speaker.z))
 			var/turf/source = speaker? get_turf(speaker) : get_turf(src)
 			src.playsound_local(source, speech_sound, sound_vol, 1)
 
-		return message
```
Comment (EgorDinamit) [inline code/modules/mob/hear_say.dm]: Why did you remove return value?

File: code/modules/mob/hear_say.dm [hunk 4]
```diff
@@ -83,22 +86,71 @@
...
-
 /mob/proc/on_hear_say(var/message)
-	to_chat(src, message)
+	to_chat(src, "[message]")
```
Comment (EgorDinamit) [inline code/modules/mob/hear_say.dm]: ..?

File: code/modules/mob/hear_say.dm [hunk 5]
```diff
@@ -83,22 +86,71 @@
...
+
+/mob/living/silicon/ai/special_mentions()
+	return list("AI") // AI door!
+
+// Converts specific characters, like *, /, and _ to formatted output.
+/mob/proc/say_emphasis(var/message)
+	message = encode_html_emphasis(message, "/", "i")
+	message = encode_html_emphasis(message, "*", "b")
+	message = encode_html_emphasis(message, "_", "u")
+	return message
```
Comment (EgorDinamit) [inline code/modules/mob/hear_say.dm:130]: Not needed, it already exists.

Comment (77seven77seven) [reply code/modules/mob/hear_say.dm:130]: what where

**Comment by 77seven77seven** [comment]
find() has been replaced by findtext

File: code/modules/mob/hear_say.dm (hunk 1 above)
Comment (77seven77seven) [reply code/modules/mob/hear_say.dm:74]: idk whats the original position and theres no reason to move it back :trollface:

Comment (EgorDinamit) [reply code/modules/mob/hear_say.dm:74]: You literally see old positions on the github. It was right under `if(language)`

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Leave a screenshot of how it looks.
Also, remove the "emphasis" stuff(bold, cursive text), as it already exists.

File: code/modules/mob/hear_say.dm (hunk 1 above)
Comment (77seven77seven) [reply code/modules/mob/hear_say.dm:74]: ok i moved em

Comment (77seven77seven) [reply code/modules/mob/hear_say.dm:74]: ok if i move them back it doesnt use the variables

**Comment by 77seven77seven** [comment]
image
